### PR TITLE
Add MW65_r2 Series PCB

### PR DESCRIPTION
**Pull request by TW59420**
## Description
MWStudio is a custom keyboard enthusiast
Upload qmk firmware of mw65_r2 series PCB

## Types of Changes
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mwstudio/mw65_r2/readme.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# MW65_R2
+
+![MW65_R2](https://i.imgur.com/QzECiuJ.jpg?2)
```
Comment (drashna) [inline keyboards/mwstudio/mw65_r2/readme.md:3]:
```suggestion
![MW65_R2](https://i.imgur.com/QzECiuJh.jpg)
```

**Review by drashna — CHANGES_REQUESTED** [review]
You also have a merge conflict that needs resolution.

**Comment by TW59420** [comment]
Please grant my request for merger, Thanks! @drashna

**Comment by drashna** [comment]
> Please grant my request for merger, Thanks! @drashna

You have merge conflicts that must be resolved still.  Until that's done, I can't actually merge it.  Also, we generally want 2 approvals before we merge.

**Comment by TW59420** [comment]
> > 请批准我的合并请求，谢谢！@德拉什纳
> 
> 您有必须仍然解决的合并冲突。在完成之前，我实际上无法合并它。此外，我们通常在合并之前需要 2 个批准。

I see,Sorry! is my problem

**Comment by TW59420** [comment]
@fauxpark  @drashna  Can you help me agree to my pull request?

File: keyboards/mwstudio/mw65_r2/config.h [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+#define MATRIX_ROWS 5
+#define MATRIX_COLS 15
+
+/* key matrix pins */
+#define MATRIX_ROW_PINS { D0, D1, D2, D3, B7 }
+#define MATRIX_COL_PINS { D6, D4, C7, C6, B6, B5, B4, D7, F7, F1, F4, F5, F6, E6, F0 }
+
+#define BOOTMAGIC_LITE_ROW 0
+#define BOOTMAGIC_LITE_COLUMN 0
+
```
Comment (drashna) [inline keyboards/mwstudio/mw65_r2/config.h:38]: These aren't really needed, if they're the default. 
```suggestion
```

File: keyboards/mwstudio/mw65_r2/config.h [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+/* Set 0 if debouncing isn't neede */
+#define DEBOUNCE 5
+
+#define RGB_DI_PIN B3
+#ifdef RGB_DI_PIN
```
Comment (drashna) [inline keyboards/mwstudio/mw65_r2/config.h:46]: No animation modes?

File: keyboards/mwstudio/mw65_r2/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    action_exec(encoder_event);
+}
+
+void matrix_scan_user(void) {
+    encoder_action_unregister();
+}
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    encoder_action_register(index, clockwise);
+    return true;
```
Comment (drashna) [inline keyboards/mwstudio/mw65_r2/keymaps/via/keymap.c:93]:
```suggestion
    return false;
```

File: keyboards/mwstudio/mw65_r2/mw65_r2.c [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "mw65_r2.h"
```
Comment (drashna) [inline keyboards/mwstudio/mw65_r2/mw65_r2.c:17]: This allows for a default behavior to be defined, without the need for the user to specify the behavior.  This is very useful for the configurator which doesn't have a way to configure this behavior, currently.

```suggestion
#include "mw65_r2.h"

#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 0) {
        if (clockwise) {
            tap_code_delay(KC_VOLU, 10);
        } else {
            tap_code_delay(KC_VOLD, 10);
        }
    }
    return true;
}
#endif
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/mwstudio/mw65_r2/rules.mk [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no      # Breathing sleep LED during USB suspend
```
Comment (fauxpark) [inline keyboards/mwstudio/mw65_r2/rules.mk:16]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mwstudio/mw65_r2/config.h (hunk 3 above)
Comment (TW59420) [reply keyboards/mwstudio/mw65_r2/config.h:43]: Just need a light band

**Review by drashna — APPROVED** [review]

**Comment by TW59420** [comment]
@fauxpark  Can you agree to my request? plase，because it's been a long time

File: keyboards/mwstudio/mw65_r2/rules.mk [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/mwstudio/mw65_r2/rules.mk:16]:
```suggestion
NKRO_ENABLE = yes           # Enable N-Key Rollover
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by TW59420** [comment]
@fauxpark i fix it

**Comment by peepeetee** [comment]
Why the PR closure? Do you need language help? I can offer to add you on QICQ

您怎么把PR关了？您需要语言帮助吗？我可以加您QQ

**Pull request closed without merging**